src/Chat/Message.tsx

```diff
@@ -1,18 +1,38 @@
 import { cn } from "@/lib/utils";
 import { ReactNode } from "react";
 import { Id } from "../../convex/_generated/dataModel";
+import { useMutation } from "convex/react";
+import { api } from "../../convex/_generated/api";
 
 export function Message({
   authorName,
   author,
   viewer,
   children,
+  id,
 }: {
   authorName: string;
   author: Id<"users">;
   viewer: Id<"users">;
   children: ReactNode;
+  id: Id<"messages">;
 }) {
+  const deleteMessage = useMutation(api.messages.deleteMessage);
+
+  const handleClick = async () => {
+    try {
+      await deleteMessage({ id });
+    } catch (error) {
+      throw new Error("Can't delete");
```

Comment:
the deleteMessage already throw the same message, why you catching and throwing it again?

Reply:
> the deleteMessage already throw the same message, why you catching and throwing it again?

My mistake!! React events expected return type as void, so I called this function inside handleDoubleClick, I should have delegated the error instead of creating a new one.